src/pages/home/library/SongList.tsx

```diff
@@ -57,26 +55,68 @@ const StyledTableCell = styled(TableCell)({
   color: theme.colors.white,
 });
 
-export default function SongList({
-  songData,
-  totalCountOfSongs,
-  currentPlayingSongId,
-  onSongPlayPause,
-  page,
-  onPageChange,
-  rowsPerPage,
-}: SongListProps) {
+export default function SongList({ totalCountOfSongs, query }: SongListProps) {
+  const navigate = useNavigate();
+  const headerHeight = 245;
+  const footerHeight = 40;
+  const bottomPadding = 30;
+  const rowHeight = 65;
+  const viewportWidth = useWindowDimensions()?.width;
+  const viewportHeight = useWindowDimensions()?.height;
+  const [rowsPerPage, setRowsPerPage] = useState(1);
+  let songsToRequest = rowsPerPage;
+  const [currentPlayingSongId, setCurrentPlayingSongId] = useState<string>();
+  const [page, setPage] = useState(1);
+
   const lastRowOnPage = (page - 1) * rowsPerPage + rowsPerPage;
+  const totalPagesCount = Math.ceil(totalCountOfSongs / rowsPerPage);
+  const remainingSongsOnLastPage = totalCountOfSongs % rowsPerPage;
 
-  // navigation for song edit page
-  const navigate = useNavigate();
+  // Determines how many songs to request for the last page
+  if (page === totalPagesCount) {
+    songsToRequest =
+      remainingSongsOnLastPage > 0 ? remainingSongsOnLastPage : rowsPerPage;
+  }
+
+  const {
+    data: songData = [],
+    isLoading,
+    isSuccess,
+  } = useGetSongsQuery({
+    ownerIds: ["me"],
+    offset: page - 1,
+    limit: songsToRequest,
+    phrase: query,
+  });
+
+  const hlsJsParams = useMemo(
+    () => ({
+      onPlaySong: ({ id }: Song) => setCurrentPlayingSongId(id),
+      onStopSong: () => setCurrentPlayingSongId(undefined),
+      onSongEnded: () => setCurrentPlayingSongId(undefined),
+    }),
+    []
+  );
```

Comment:
So i went back and forth on this, but ultimately chose to go this route bc we needed the search to be part of a parent component that tells the children what to render/fetch otherwise if we did it all in the child, it would trigger a component rerender and would take the active focus element off and cause unnecessary rerenders. If there's a better approach i am opened to it

Reply:
Is that part of the reason you keep the `totalCountOfSongs` in the parent `Discography.tsx` or would that be able to come down to `SongList.tsx` as well since it just needs `query`?

Reply:
Yup that's why, otherwise the count was going to be off